uaclient/entitlements/fips.py

```diff
@@ -257,11 +258,20 @@ def install_packages(
                     ],
                 )
             except exceptions.UbuntuProError:
-                event.info(
+                progress.emit(
+                    "info",
                     messages.FIPS_PACKAGE_NOT_AVAILABLE.format(
                         service=self.title, pkg=pkg
-                    )
+                    ),
                 )
+        if self._check_for_reboot():
+            notices.add(
+                Notice.FIPS_SYSTEM_REBOOT_REQUIRED,
+            )
+
+    def _check_for_reboot(self) -> bool:
```

Comment:
Why are we overwriting this method here ? Since it is exactly the same as the one defined on the base class, shouldn't we instead adapt this method on the `repo` class instead ?

Reply:
I changed the implementation in the base class :grimacing: let me know what you think.

Reply:
I missed that on the review :facepalm: But why are we returning False by default on the base class now ? It is just because the logic is redundant considering the implementation we have on the repo class ?

Reply:
We have agreed that the necessary implementation here should be on the repo class (which already correctly implements the logic)